### DISCUSSION TITLE
Rasterizing overprints/spot colors with opacity

### DISCUSSION THREAD

**Discussion by mcdado** [libvips/libvips#3420] [General]
Hi! I recently came to know libvips and it's very promising for some use cases that we have.
We are looking for a performant way to rasterize PDFs for print, so I tried both vips copy, vips thumbnail, etc, but fundamentally I would love to interface with vips via Node.js with [sharp](https://github.com/lovell/sharp) — at our shop we don't write in low level language like C++ but if anybody can provide some tips on how to use the library, maybe we can take some cues from them.

We are having some issues with colors: when we have some elements in overprint, we cannot get the raster image to simulate the overprint correctly. Here is an example file: [Download test PDF](https://github.com/libvips/libvips/files/11109511/test3.pdf)

As reference, here is a screenshot of Acrobat's output preview with the option to simulate overprint enabled. This would be the result that we want in rasterized form:
![PDF in Acrobat, expected result](https://user-images.githubusercontent.com/898057/228777052-46d6f9ac-b126-4866-ae21-878ed7a1420d.png)

When we tried to use vips thumbnail feature, we got weird results. We tried several ways to configure the color profile via the command line (various vips commands) but we never got it to work:
![PNG generated by vips thumbnail, with bad colors](https://user-images.githubusercontent.com/898057/228777438-63bf5472-017c-44f4-bad1-1d0ceae44471.png)

The closer we got to the desired result was to use ImageMagick converting the color profile from FOGRA39 to RGB, something that we didn't manage to do with libvips:
![PNG generated by ImageMagick, close to desired result](https://user-images.githubusercontent.com/898057/228778201-9e761e84-e576-4386-9f43-9c35336eea0b.png)

With ImageMagick there's a problem with spot colors in overprint, also it is quite slow to process even a single file like this. We are looking at libvips because it seems like a more performant and more versatile tool. Can somebody guide us on how to achieve the result we're looking for? Thank you in advance.

**Comment by jcupitt** [comment]
Hi @mcdado,

Sorry, I doubt if this is possible. libvips uses either poppler (the library xpdf uses for PDF viewing) or pdfium (the thing Chrome uses for PDF viewing) for PDF rendering, and both libraries only do RGB PDFs (they are designed around screen output).

ImageMagick handles PDFs by calling out to GhostScript. This is a much more sophisticated (and much slower!) system that can handle CMYK correctly, and I'd expect GhostScript to be able to simulate overprints if you configure it. I don't know if imagemagick has the controls you need to be able to set this up correctly though.

I would look into invoking ghostscript directly, and read up on overprint support. 

Be aware that GS has quite a restrictive license and you might need to pay for it. 

https://www.ghostscript.com/licensing

I imagine it would still be cheaper than the $lol amount Adobe want.

**Comment by mcdado** [comment]
Thank you @jcupitt, that was very informative.
